x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -201,267 +208,379 @@ export class AnalyticsManager {
+  async extendAnalyticsMapForAnalyticsJob({
+    analyticsId,
+    index,
+  }: {
+    analyticsId?: string;
+    index?: string;
+  }): Promise<AnalyticsMapReturnType> {
     const result: any = { elements: [], details: {}, error: null };
-
     try {
       await this.setInferenceModels();
+      const jobs = await this.getAnalyticsData();
+      let rootIndex;
+      let rootIndexNodeId;
+
+      if (analyticsId !== undefined) {
+        const jobData = await this.getAnalyticsData(analyticsId);
+        const currentJobNodeId = `${jobData.id}-${JOB_MAP_NODE_TYPES.ANALYTICS}`;
+        rootIndex = Array.isArray(jobData?.dest?.index)
+          ? jobData?.dest?.index[0]
+          : jobData?.dest?.index;
+        rootIndexNodeId = `${rootIndex}-${JOB_MAP_NODE_TYPES.INDEX}`;
+
+        // Fetch inference model for incoming job id and add node and edge
+        const { modelElement, modelDetails, edgeElement } = this.getAnalyticsModelElements(
+          analyticsId
+        );
+        if (isAnalyticsMapNodeElement(modelElement)) {
+          result.elements.push(modelElement);
+          result.details[modelElement.data.id] = modelDetails;
+        }
+        if (isAnalyticsMapEdgeElement(edgeElement)) {
+          result.elements.push(edgeElement);
+        }
 
-      const jobData = await this.getAnalyticsJobData(analyticsId);
-      const currentJobNodeId = `${jobData.id}-${JOB_MAP_NODE_TYPES.ANALYTICS}`;
-      const destIndex = Array.isArray(jobData?.dest?.index)
-        ? jobData?.dest?.index[0]
-        : jobData?.dest?.index;
-      const destIndexNodeId = `${destIndex}-${JOB_MAP_NODE_TYPES.INDEX}`;
-      const analyticsJobs = await this._mlClient.getDataFrameAnalytics();
-      const jobs = analyticsJobs?.body?.data_frame_analytics || [];
-
-      // Fetch inference model for incoming job id and add node and edge
-      const { modelElement, modelDetails, edgeElement } = this.getAnalyticsModelElements(
-        analyticsId
-      );
-      if (isAnalyticsMapNodeElement(modelElement)) {
-        result.elements.push(modelElement);
-        result.details[modelElement.data.id] = modelDetails;
-      }
-      if (isAnalyticsMapEdgeElement(edgeElement)) {
-        result.elements.push(edgeElement);
+        // If rootIndex node has not been created, create it
+        const rootIndexDetails = await this.getIndexData(rootIndex);
+        result.elements.push({
+          data: {
+            id: rootIndexNodeId,
+            label: rootIndex,
+            type: JOB_MAP_NODE_TYPES.INDEX,
+          },
+        });
+        result.details[rootIndexNodeId] = rootIndexDetails;
+
+        // Connect incoming job to rootIndex
+        result.elements.push({
+          data: {
+            id: `${currentJobNodeId}~${rootIndexNodeId}`,
+            source: currentJobNodeId,
+            target: rootIndexNodeId,
+          },
+        });
+      } else {
+        rootIndex = index;
```

Comment:
Seems like we are assuming if `analyticsId` is not defined, then `index` will be. Suggestion to make the type of the input arguments more explicit or add a check here.